demos/smart_classroom_demo/src/action_detector.cpp

```diff
@@ -131,26 +131,35 @@ DetectedActions ActionDetection::fetchResults() {
     const auto loc_blob_name = new_network_ ? config_.new_loc_blob_name : config_.old_loc_blob_name;
     const auto det_conf_blob_name = new_network_ ? config_.new_det_conf_blob_name : config_.old_det_conf_blob_name;
 
-    const cv::Mat priorbox_out =
-        new_network_
-          ? cv::Mat()
-          : cv::Mat(ieSizeToVector(request->GetBlob(config_.old_priorbox_blob_name)->getTensorDesc().getDims()),
-                    CV_32F, request->GetBlob(config_.old_priorbox_blob_name)->buffer());
-
+    LockedMemory<const void> locBlobMapped = as<MemoryBlob>(request->GetBlob(loc_blob_name))->rmap();
     const cv::Mat loc_out(ieSizeToVector(request->GetBlob(loc_blob_name)->getTensorDesc().getDims()),
-                          CV_32F, request->GetBlob(loc_blob_name)->buffer());
+                          CV_32F, locBlobMapped.as<float*>());
 
+    LockedMemory<const void> detConfBlobMapped = as<MemoryBlob>(request->GetBlob(det_conf_blob_name))->rmap();
     const cv::Mat main_conf_out(ieSizeToVector(request->GetBlob(det_conf_blob_name)->getTensorDesc().getDims()),
-                                CV_32F, request->GetBlob(det_conf_blob_name)->buffer());
+                                CV_32F, detConfBlobMapped.as<float*>());
 
     std::vector<cv::Mat> add_conf_out;
+    std::vector<LockedMemory<const void>> blobsMapped;
```

Comment:
I would suggest to swap these lines. It makes no practical difference, but it's better style-wise, because:

1. you add elements to `blobsMapped` before `add_conf_out`, and
1. since objects are destroyed in reverse order, there would no longer be a moment at the end of the function when the `Mat`s are alive but the `LockedMemory`s they are referring to are not.